### ISSUE TITLE
Saving Rules to the wrong Site

### ISSUE THREAD

**Issue by erickgonzalez** [dotCMS/core#11461]
## Expected Behavior
If you create a new Site and add a rule, the rule should be saved on that site.

## Current Behavior
If you create a new Site and add a rule, the rule is saved on demo.dotcms.com site.

## Steps to Reproduce (for bugs)
1. Create a new site
2. On the Site Selector, select the new site
3. Go to Rules Portlet
4. Create a new Rule
5. Go to any other portlet
6. Back at the Rules Portlet
7. The Rule created is not showing

If you open the dev tools, on the Network tab, you can see that Rule is being saved on the demo site, regardless the site selected was other.
```
Request URL:https://demo.dotcms.com/api/v1/sites/48190c8c-42c4-46af-8d1a-0cd5db894797/ruleengine/rules/137fa880-3e85-4005-97ee-c9456909b307
```
Also if you select the demo.dotcms.com site and go to the Rules Portlet you can see the created Rule.

## Your Environment
* dotCMS version used: 4.0.1 and current master
* Browser Name and version: FF and Chrome
* Database and version: Postgres

Client's Ticket: https://my.dotcms.com/tickets/detail.dot?id=ebddc1b2-b2b0-446e-a1fb-3480fedf5b50

**Comment by oswaldogallango** [comment]
P.R https://github.com/dotCMS/core-web/pull/347

**Comment by oswaldogallango** [comment]
P.R to master branch https://github.com/dotCMS/core-web/pull/348

**Comment by bryanboza** [comment]
Fixed, tested on release-4.1 // Chrome  and after upgrade from 4.0 and works as expected...

**Comment by wezell** [comment]
So, on 3.x, the iframe that loaded rules passed a realmId variable that said what page or site the rule should be on.  It looks like we have lost this in 4.1:

https://demo3.dotcms.com/html/js/_rulesengine/?locale=en-US&realmId=48190c8c-42c4-46af-8d1a-0cd5db894797&hideFireOn=true&hideRulePushOptions=false

**Comment by bryanboza** [comment]
Fixed, another bug open in the #11585
